test/e2e/metrics_test.go

```diff
@@ -0,0 +1,43 @@
+package e2e
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	core "k8s.io/api/core/v1"
+)
+
+// testMetrics deploys a Linux pod and tests that it can communicate with the Windows node's metrics port
+func testMetrics(t *testing.T) {
+	testCtx, err := NewTestContext(t)
+	require.NoError(t, err)
+
+	// Need at least one Windows node to run these tests, throwing error if this condition is not met
+	require.GreaterOrEqualf(t, len(gc.nodes), 1, "insufficient number of Windows nodes to run tests across"+
+		" nodes, Minimum node count: 1, Current node count: %d", len(gc.nodes))
+
+	for _, winNode := range gc.nodes {
+		t.Run(winNode.Name, func(t *testing.T) {
+			// Get the node internal IP so we can curl it
+			var winNodeInternalIP string
+			for _, address := range winNode.Status.Addresses {
+				if address.Type == core.NodeInternalIP {
+					winNodeInternalIP = address.Address
+				}
+			}
+
```

Comment:
You need to check if `winNodeInternalIP` is set here.

Reply:
If the value is not set (meaning the windows VM does not have an internal IP associated with it), would we just skip testing metrics or something more drastic like erroring out?

Reply:
I lean towards skipping in this scenario, assuming there's some use case for nodes without internal IPs

Reply:
I don't think there is any use case for Windows node without internal IPs. This needs to be a requirement for a test to run and cannot be silently skipped like you are doing now.

/approve cancel